packages/devtools-source-map/src/source-map.js

```diff
@@ -182,6 +184,108 @@ async function getOriginalLocation(location: Location) : Promise<Location> {
   };
 }
 
+function _compareLocation(a: Location, b: Location): number {
+  if (a.line === b.line) {
+    return (a.column || 0) - (b.column || 0);
+  } else {
+    return a.line - b.line;
+  }
+}
+
+async function getLocationScopes(
+  location: Location,
+  generatedSourceScopes: SourceScope[]
+) : Promise<MappedScopeBindings[]|null> {
+  if (!isGeneratedId(location.sourceId)) {
+    return null;
+  }
+
+  const map = await _getSourceMap(location.sourceId);
+  if (!map) {
+    return null;
+  }
+
+  return generatedSourceScopes.map(scope => {
+    // The scope has bindings that contains positions of reference to the
+    // generated name (that may has different original name). Going over all
+    // these positions, check what original name is listed there. To make things
+    // more interesting the same generated name can potentially be used for
+    // different original names.
+    let trackedBindings: { [string]:
```

Comment:
This type should be defined above:

`let trackedBindings: TrackedBindings`

Reply:
done